File: src/Functional/OCaml/bridge.ml
type device = {
  power_on : unit -> string;
  mute : unit -> string;
}

let make_device name =
  {
    power_on = (fun () -> name ^ ":on");
    mute = (fun () -> name ^ ":muted");
  }

let activate_basic device = device.power_on ()
let activate_mute device = device.mute ()

let () =
  let tv = make_device "TV" in
  let radio = make_device "Radio" in
  print_endline ("basic-tv=" ^ activate_basic tv);
  print_endline ("basic-radio=" ^ activate_basic radio);
  print_endline ("mute-tv=" ^ activate_mute tv);
  print_endline ("mute-radio=" ^ activate_mute radio)

File: src/Functional/Erlang/bridge.erl
-module(bridge).
-export([main/0]).

make_device(Name) ->
    #{power_on => fun() -> Name ++ ":on" end,
      mute => fun() -> Name ++ ":muted" end}.

activate_basic(Device) ->
    PowerOn = maps:get(power_on, Device),
    PowerOn().

activate_mute(Device) ->
    Mute = maps:get(mute, Device),
    Mute().

main() ->
    Tv = make_device("TV"),
    Radio = make_device("Radio"),
    io:format("basic-tv=~s~n", [activate_basic(Tv)]),
    io:format("basic-radio=~s~n", [activate_basic(Radio)]),
    io:format("mute-tv=~s~n", [activate_mute(Tv)]),
    io:format("mute-radio=~s~n", [activate_mute(Radio)]).
